src/BaGetter.Web/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -28,8 +30,40 @@ public static class IServiceCollectionExtensions
         services.AddTransient<IUrlGenerator, BaGetterUrlGenerator>();
 
         services.AddSingleton(ApplicationVersionHelper.GetVersion());
-        services.AddBaGetterApplication(configureAction);
+
+        var app = services.AddBaGetterApplication(configureAction);
+        app.AddNugetBasicHttpAuthentication();
+        app.AddNugetBasicHttpAuthorization();
 
         return services;
     }
+
+    private static BaGetterApplication AddNugetBasicHttpAuthentication(this BaGetterApplication app)
+    {
+        app.Services.AddAuthentication(options =>
+        {
+            // Breaks existing tests if the contains check is not here.
+            if (!options.SchemeMap.ContainsKey(AuthenticationConstants.NugetBasicAuthenticationScheme))
+            {
+                options.AddScheme<NugetBasicAuthenticationHandler>(AuthenticationConstants.NugetBasicAuthenticationScheme, AuthenticationConstants.NugetBasicAuthenticationScheme);
+                options.DefaultAuthenticateScheme = AuthenticationConstants.NugetBasicAuthenticationScheme;
+                options.DefaultChallengeScheme = AuthenticationConstants.NugetBasicAuthenticationScheme;
+            }
+        });
+
+        return app;
+    }
+
+    private static BaGetterApplication AddNugetBasicHttpAuthorization(this BaGetterApplication app)
+    {
+        app.Services.AddAuthorization(options =>
```

Comment:
If I want to add BaGetter to my application and use multiple policies, would it work with this? Can I `.AddAuthorization()` with my policy and then call this for the nuget feed and they stack? If not, is there a way to make them stack (I can only think of taking an optional `Action<options>` parameter and call it along with `.AddPolicy`.

Reply:
Moved